### PR TITLE
Schema Usage support

### PR DESCRIPTION
**Pull request by bbakerman**
A PR on getting schema usage stats and "reference" ability

### CODE REVIEW

File: src/main/java/graphql/Directives.java [hunk 1]
```diff
@@ -124,4 +127,14 @@ private static Description createDescription(String s) {
     }
 
 
+    private final static Set<String> SPECIFIED_DIRECTIVES = ImmutableSet.of(
```
Comment (andimarek) [inline src/main/java/graphql/Directives.java:130]: we have the same methods/logic already in DirectiveInfo I believe.

File: src/main/java/graphql/schema/usage/SchemaUsage.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+            }
+        }
+        missCache.add(elementName);
+        return false;
+    }
+
+    @Internal
+    static class Builder {
+        Map<String, Integer> fieldReferenceCounts = new HashMap<>();
+        Map<String, Integer> inputFieldReferenceCounts = new HashMap<>();
```
Comment (andimarek) [inline src/main/java/graphql/schema/usage/SchemaUsage.java:232]: I would recommend ImmutableMaps.Builder here, which preserve the order and are immutable OR our beloved `LinkedHashMap` or a comment why the order is never interesting.

**Review by andimarek — CHANGES_REQUESTED** [review]

File: src/main/java/graphql/Directives.java (hunk 1 above)
Comment (bbakerman) [reply src/main/java/graphql/Directives.java:130]: nice

File: src/main/java/graphql/schema/usage/SchemaUsage.java (hunk 2 above)
Comment (bbakerman) [reply src/main/java/graphql/schema/usage/SchemaUsage.java:232]: ImmutableMap.Builder does not support `compute` like behavior.  I want to accumulate values (counts) against type names.

```
            private BiFunction<String, Integer, Integer> incCount() {
                return (k, v) -> v == null ? 1 : v + 1;
            }

builder.fieldReferenceCounts.compute(fieldType.getName(), incCount());
```

With map builder it has only PUT and no read capability for past values

So I cant use it.  I do use it on getters however.  I will improve it to create these once and once only

**Pull request merged**